crux-lucene/src/crux/lucene.clj

```diff
@@ -225,20 +227,28 @@
    ::sys/before #{[:crux/tx-ingester]}}
   [{:keys [^Path db-dir index-store document-store bus analyzer indexer query-engine] :as opts}]
   (let [directory (FSDirectory/open db-dir)
-        lucene-store (LuceneNode. directory analyzer indexer)]
+        index-writer (->index-writer directory analyzer)
+        searcher-manager (SearcherManager. index-writer false false nil)
+        lucene-store (LuceneNode. directory analyzer index-writer searcher-manager indexer)]
     ;; Ensure lucene index exists for immediate queries:
-    (with-open [index-writer (index-writer lucene-store)]
-      (.commit index-writer))
+    (.commit index-writer)
     (validate-lucene-store-up-to-date index-store lucene-store)
     (q/assoc-pred-ctx! query-engine ::lucene-store lucene-store)
     (bus/listen bus {:crux/event-types #{:crux.tx/committing-tx}
                      :crux.bus/executor (reify java.util.concurrent.Executor
                                           (execute [_ f]
                                             (.run f)))}
                 (fn [ev]
-                  (with-open [index-writer (index-writer lucene-store)]
-                    (index! indexer index-writer (db/fetch-docs document-store (:doc-ids ev)))
-                    (when-let [evicting-eids (not-empty (:evicting-eids ev))]
-                      (evict! indexer index-writer evicting-eids))
-                    (index-tx! index-writer (:submitted-tx ev)))))
+                  (index! indexer index-writer (db/fetch-docs document-store (:doc-ids ev)))
+                  (when-let [evicting-eids (not-empty (:evicting-eids ev))]
+                    (evict! indexer index-writer evicting-eids))
+                  (index-tx! index-writer (:submitted-tx ev))
+                  (.commit index-writer)
+                  ;; it's _far_ less than ideal to .commit here per transaction
```

Comment:
I would remove this comments and place them in this GH issue. We don't know there's a valid alternative to calling `commit` on each tx. and if users partition the tx-ops into txes, then it's not potentially an issue at all (subject to your microbench results)

Reply:
I think the need to avoid a `commit` per tx might be more apparent at much larger scales, and particularly when `:sync false` is set on the KV index-store. The solution shouldn't be too hard - I think we just need some kind of background service thread that calls `commit` followed by `fsync` on the KV protocol.